### PR TITLE
[5.x] Add prometheus metrics to pg and redis operations

### PR DESCRIPTION
**Pull request by vangheem**

### CODE REVIEW

**Review by masipcat — COMMENTED** [review]
The idea is to keep guillotina_prometheus to expose these metrics?

File: guillotina/contrib/redis/dm.py [hunk 1]
```diff
@@ -42,7 +69,8 @@ async def _save(self):
         redis = await self.get_redis()
         key = self.get_key()
         self._data["last_activity"] = time.time()
-        await redis.set(key, json.dumps(self._data, cls=GuillotinaJSONEncoder), expire=self._ttl)
+        with watch("set"):
+            await redis.set(key, json.dumps(self._data, cls=GuillotinaJSONEncoder), expire=self._ttl)
```
Comment (masipcat) [inline guillotina/contrib/redis/dm.py:73]: Should the `json.dumps` be outside the contextmanager?

**Review by bloodbare — APPROVED** [review]
Thanks!

**Comment by vangheem** [comment]
> The idea is to keep guillotina_prometheus to expose these metrics?

Yes, or you can expose them yourself by implementing your own metrics endpoint.

File: guillotina/contrib/redis/dm.py (hunk 1 above)
Comment (vangheem) [reply guillotina/contrib/redis/dm.py:73]: good idea. I will move it

**Review by masipcat — APPROVED** [review]

**Comment by masipcat** [comment]
No port for G6? I'd like to get this metrics!

**Pull request merged**